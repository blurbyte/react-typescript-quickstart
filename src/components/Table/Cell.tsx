import styled from '@styles/styledComponents';
import { Colors } from '@styles/colors';

const Cell = styled.td`
  font-family: Helvetica, Arial, sans-serif;
  text-align: left;
  padding: 1.4rem;

  &:first-child {
    border-right: 1px solid ${Colors.gray};
  }

  & strong {
    font-weight: 700;
  }
`;

export default Cell;
