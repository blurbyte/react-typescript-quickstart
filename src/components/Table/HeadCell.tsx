import styled from '@styles/styledComponents';

import BaseCell from './Cell';

const Cell = BaseCell.withComponent('th');

const HeadCell = styled(Cell)`
  font-family: 'Open Sans', Helvetica, Arial, sans-serif;
  font-weight: 300;
  background: rgba(40, 40, 40, 0.1);
`;

export default HeadCell;
