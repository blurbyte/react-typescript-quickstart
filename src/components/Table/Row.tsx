import styled from 'styled-components';

const Row = styled.tr`
  background: rgba(40, 40, 40, 0.02);
  transition: all 0.2s ease;

  &:hover {
    background: rgba(74, 108, 187, 0.2);
  }
`;

export default Row;
