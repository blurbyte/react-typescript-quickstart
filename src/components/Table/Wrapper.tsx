import styled from '@styles/styledComponents';
import { Color } from '@styles/colors';

const Wrapper = styled.table`
  border-top: 2px solid ${Color.Gray};
  border-collapse: collapse;
  color: ${Color.DarkGray};
  margin: 1rem 0;
  width: 100%;
`;

export default Wrapper;
