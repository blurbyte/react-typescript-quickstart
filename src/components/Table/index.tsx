import * as React from 'react';

import Cell from './Cell';
import HeadCell from './HeadCell';
import Row from './Row';
import Wrapper from './Wrapper';

export interface TableProps {
  children?: React.ReactNode;
}

class Table extends React.Component<TableProps, {}> {
  static Cell = Cell;
  static HeadCell = HeadCell;
  static Row = Row;

  render() {
    return <Wrapper>{this.props.children}</Wrapper>;
  }
}

export default Table;
