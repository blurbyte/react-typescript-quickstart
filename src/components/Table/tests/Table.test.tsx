import * as React from 'react';
import { shallow } from 'enzyme';

import Table from '../index';

describe('Table', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Table />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly Cell', () => {
    const wrapper = shallow(<Table.Cell />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly HeadeCell', () => {
    const wrapper = shallow(<Table.HeadCell />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly Row', () => {
    const wrapper = shallow(<Table.Row />);
    expect(wrapper).toMatchSnapshot();
  });
});
