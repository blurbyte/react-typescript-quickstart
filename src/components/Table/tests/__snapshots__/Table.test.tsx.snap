// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `<styled.table />`;

exports[`Table renders correctly Cell 1`] = `
.c0 {
  font-family: Helvetica,Arial,sans-serif;
  text-align: left;
  padding: 1.4rem;
}

.c0:first-child {
  border-right: 1px solid #908e89;
}

.c0 strong {
  font-weight: 700;
}

<td
  className="c0"
/>
`;

exports[`Table renders correctly HeadeCell 1`] = `
.c0 {
  font-family: 'Open Sans',Helvetica,Arial,sans-serif;
  font-weight: 300;
  background: rgba(40,40,40,0.1);
}

<styled.th
  className="c0"
/>
`;

exports[`Table renders correctly Row 1`] = `
.c0 {
  background: rgba(40,40,40,0.02);
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c0:hover {
  background: rgba(74,108,187,0.2);
}

<tr
  className="c0"
/>
`;
